### PR TITLE
RUBY-1362 Implement SASLprep for Ruby versions below 2.2

### PR DESCRIPTION
**Pull request by saghm**
This PR also gets rid of the messy testing conditionals that were added as part of the work for implementing SCRAM-SHA-256, as they are no longer necessary after this change.

### CODE REVIEW

File: lib/mongo/auth/stringprep.rb [hunk 1]
```diff
@@ -104,7 +96,12 @@ def mapping(c, mappings)
       end
 
       def normalize!(out)
-        out.unicode_normalize!(:nfkc)
+        if RUBY_VERSION < '2.2.0'
+          require 'mongo/auth/stringprep/unicode_normalize/normalize'
```
Comment (p-mongo) [inline lib/mongo/auth/stringprep.rb]: You may consider only requiring this if `::UnicodeNormalize` is not an existing constant.

**Comment by saghm** [comment]
This was taken out of the latest stable, 2.5.1

File: THIRD-PARTY-NOTICES [hunk 2]
```diff
@@ -0,0 +1,64 @@
+------------------------------------------------------------------------
```
Comment (behackett) [inline THIRD-PARTY-NOTICES]: Use the format found here:

https://github.com/mongodb/mongo-python-driver/blob/master/THIRD-PARTY-NOTICES#L1-L16

As specified here:

https://jira.mongodb.org/browse/DRIVERS-323

Including specifying what files the license applies to.

File: lib/mongo/auth/stringprep.rb (hunk 1 above)
Comment (behackett) [reply lib/mongo/auth/stringprep.rb]: Good idea. That covers alternative implementations that might add it in a different version. For example, does RUBY_VERSION < '2.2.0' also cover JRuby?

**Comment by saghm** [comment]
Updated for feedback

**Pull request merged**